### ISSUE TITLE
I wish for flutter doctor printed a one-line summary at the very end

### ISSUE THREAD

**Issue by sethladd** [flutter/flutter#12767]
flutter doctor has a lot of awesome output, but it's easy to miss something important. Users will want to know, in a single line, "am I good to go?"

Users don't always have the ability to see all the output. here's an example of VS Code:

<img width="743" alt="screen shot 2017-10-27 at 11 02 26 am" src="https://user-images.githubusercontent.com/5479/32118458-a62b414c-bb06-11e7-9b1c-66a389ef4910.png">

Notice that only the bottom third of the output is visible.

Here's one idea: at the very end of the doctor output, we should show:

"You're good to go! Flutter is ready for Android [and iOS] development"

or

"Sorry, we need your help. One or more prereqs need attention before Flutter is good to go."

**Comment by eseidelGoogle** [comment]
As Flutter Doctor gets more complicated, may want to make it more like `brew doctor` which only prints details on failure.  Our current output is more of a "what is my install overview" question rather than an "am I OK?" question.

```
% brew doctor
Your system is ready to brew.
```

**Comment by sethladd** [comment]
sgtm

for bug reports, we could ask for `flutter --version` which lists everything.

**Comment by b-strauss** [comment]
There's already a `flutter --version`.

**Comment by franzsilva** [comment]
Maybe 
`flutter doctor`
Could be the single line. 
And 
`flutter doctor --verbose`
Gives you the entire output.

**Comment by mit-mit** [comment]
I think our "definition" of "failing" is a bit more subtle. For example, I might fail the iOS checks but not care about it if currently focused on Android, and thus the difference between when you see very little output and a lot of output becomes fuzzy.

I think it would be better if `flutter doctor` reported just a single line for each category, for example:

```
[✓] Flutter (on Mac OS X 10.13.2 17C88, locale en-DK, channel master)
[✓] Android toolchain (Android SDK 26.0.3)
[✓] iOS toolchain (Xcode 9.2)
[✓] Android Studio (version 3.0)
[✓] IntelliJ IDEA Community Edition (version 2017.3.2)
[✓] 2 connected devices
```

And:

```
[✓] Flutter (on Mac OS X 10.13.2 17C88, locale en-DK, channel master)
[✓] Android toolchain (Android SDK 26.0.3)
[x] iOS toolchain (Xcode 9.2)
[✓] Android Studio (version 3.0)
[✓] IntelliJ IDEA Community Edition (version 2017.3.2)
[✓] 2 connected devices

Doctor found issues 1 category; run 'flutter doctor -v' for details.
```

`flutter doctor --verbose` prints all the current details.

**Comment by mit-mit** [comment]
Tagging first-hour and milestone 3 as this impacts new users the most.

**Comment by DanTup** [comment]
If work is being done on this, it might be efficient to do #10621 at the same time :D

**Comment by DanTup** [comment]
.. and I guess doing #10352 would make it most useful; it'll be a bit weird of VS Code users are told to install Android Studio and/or IntelliJ. I guess that might be what @mit-mit's comments are about above, but specifically being able to separate IDEs from other optional components (if there are any) would be handy.

**Comment by mit-mit** [comment]
@DanTup those linked bugs are a bit more complicated, and not currently in scope for milestone 3 (specifically, we currently do depend on a full Android Studio install incl. all the SDK content it vends for us).

**Comment by DanTup** [comment]
> those linked bugs are a bit more complicated, and not currently in scope for milestone 3 

Understood. I don't know the details of how these work but thought it was worth mentioning in case they were easy wins :-)

> specifically, we currently do depend on a full Android Studio install incl. all the SDK content it vends for us

I don't currently have Android Studio installed, but I have working Flutter on both Windows and Mac. It was a slight faff (getting the Android SDK and knowing which packages to download) but I don't think anything that couldn't be easily documented. (I don't know if it's huge value - I personally like to minimise what's installed on my machine ,so wanted to avoid installing an IDE I wasn't using, but I doubt this is typical user behaviour).

**Comment by jcollins-g** [comment]
While probably outside the scope of this issue, one possibility might be to have flutter doctor "pass" as long as at least one SDK (Android or iOS) is present and working, but fail if we detect the situation where one works and the other isn't working (e.g. needs license accepted) or on any situation where we don't have one at all.  This way the tool is more likely to match the user's intent -- I tend to be like @DanTup and not install things I don't think I need.  I did the recent flutter hackathon without Android Studio installed at all.

**Comment by mit-mit** [comment]
Yes, I think that would work. Let's try that (I guess it would be easy to adjust later if it turns out to still be too noisy)

**Comment by jcollins-g** [comment]
Fixed in #14173.
